python/chemiscope/adapters.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+
+try:
+    import ase
+
+    HAVE_ASE = True
+except ImportError:
+    HAVE_ASE = False
+
+
+def frames_to_json(frames):
+    frames = list(frames)
+
+    if HAVE_ASE and isinstance(frames[0], ase.Atoms):
+        return [_ase_to_json(frame) for frame in frames]
+    else:
+        raise Exception(f"unknown frame type: '{frames[0].__class__.__name__}'")
+
```

Comment:
To improve readability, this function should be at the end of the file.

Reply:
I prefer having the main function on the top of the file, and the specific implementation further down. This way the main part if visible first when opening the file.